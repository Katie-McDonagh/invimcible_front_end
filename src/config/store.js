import { createStore, combineReducers } from 'redux'
import playerReducer from '../features/player/reducer'
import mapReducer from '../features/map/reducer'
import vimCommandReducer from '../components/vimCommands/reducer'

const rootReducer = combineReducers({
  player: playerReducer,
  map: mapReducer,
  vimCommand: vimCommandReducer
})

const store = createStore(
  rootReducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
)

export default store
