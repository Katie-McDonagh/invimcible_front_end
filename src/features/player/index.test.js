import React from "react";
import Player from "./index.js";
import { shallow, mount } from "enzyme";
import { Provider } from "react-redux";
import store from "../../config/store";
import renderer from "react-test-renderer";

let component

beforeEach(() => {
  component = renderer.create(
    <Provider store={store}>
      <Player />
    </Provider>
  );
});

describe("Map", () => {
  it("renders map information", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
})
