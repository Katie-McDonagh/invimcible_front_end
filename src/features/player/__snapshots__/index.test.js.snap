// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map renders map information 1`] = `
<div
  style={
    Object {
      "backgroundImage": "url('player_walk.png')",
      "backgroundPosition": "0px 0px",
      "height": "32px",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "32px",
    }
  }
/>
`;
